std/format.d

```diff
@@ -5456,12 +5477,41 @@ private bool needToSwapEndianess(Char)(const ref FormatSpec!Char f)
     assert(stream.data == "C");
 }
 
+// Used to check format strings are compatible with argument types
+package static const checkFormatException(alias fmt, Args...) =
+{
+    try
+        .format(fmt, Args.init);
+    catch (Exception e)
+        return (e.msg is ctfpMessage) ? null : e;
+    return null;
+}();
+
 /*****************************************************
  * Format arguments into a string.
  *
- * Params: fmt  = Format string. For detailed specification, see $(REF formattedWrite, std,_format).
- *         args = Variadic list of arguments to format into returned string.
+ * Params: fmt  = Format string. For detailed specification, see $(LREF formattedWrite).
+ *         args = Variadic list of arguments to _format into returned string.
  */
+typeof(fmt) format(alias fmt, Args...)(Args args)
```

Comment:
Might be a good idea to add sig constraints here to constrain what the user passes in `fmt`.